ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/transactions/PendingTransactions.java

```diff
@@ -52,8 +52,9 @@
   private final Map<Hash, TransactionInfo> pendingTransactions = new HashMap<>();
   private final SortedSet<TransactionInfo> prioritizedTransactions =
       new TreeSet<>(
-          comparing(TransactionInfo::isReceivedFromLocalSource)
+          comparing(PendingTransactions.TransactionInfo::isReceivedFromLocalSource)
               .thenComparing(TransactionInfo::getSequence)
+              .thenComparing(txInfo -> txInfo.getTransaction().getGasPrice())
```

Comment:
I think this will still remove the tx with the highest nonce, then use gas as a tie-breaker.